modules/storages/app/views/storages/admin/storages/_form.html.erb

```diff
@@ -33,17 +33,25 @@ See COPYRIGHT and LICENSE files for more details.
 <fieldset class="form--fieldset">
   <legend class="form--fieldset-legend"><%= I18n.t(:label_general) %></legend>
   <div class="form--field -required">
-    <%= f.hidden_field :provider_type %>
-    <%= f.select :provider_type,
-                 ::Storages::Storage::PROVIDER_TYPES.map { |provider_type| [I18n.t("storages.provider_types.#{Storages::Storage.shorten_provider_type(provider_type)}.name"), provider_type] },
-                 {
-                   selected: ::Storages::Storage::PROVIDER_TYPE_NEXTCLOUD,
-                   container_class: '-slim'
-                 },
-                 {
-                   disabled: @object.persisted? || ::Storages::Storage::PROVIDER_TYPES.count == 1
-                 } %>
-    <span class="form--field-instructions"><%= t('storages.instructions.type') %> <%= t('storages.instructions.type_link_text') %></span>
+    <% case action_name %>
+    <% when 'new' %>
+      <%= f.hidden_field :provider_type %>
+      <%= f.select :provider_type,
+                  ::Storages::Storage::PROVIDER_TYPES.map { |provider_type| [I18n.t("storages.provider_types.#{::Storages::Storage.shorten_provider_type(provider_type)}.name"), provider_type] },
+                  {
+                    selected: ::Storages::Storage::PROVIDER_TYPE_NEXTCLOUD,
+                    container_class: '-slim'
+                  }
+      %>
+      <span class="form--field-instructions"><%= t('storages.instructions.type') %> <%= t('storages.instructions.type_link_text') %></span>
+    <% when 'edit' %>
+      <%= f.label :provider_type, class: "form--label" do %>
+        <%= t(:"storages.label_provider_type") %>
+      <% end %>
+      <div class="form--field-container -horizontal">
+        <span><%= @object.name %></span>
+      </div>
+    <% end %>
```

Comment:
Was this discussed, that we disable switching the storage type when editing a storage?

Reply:
Yes, we discussed this with @wielinde and thought to make it consitent with the project settings form. Do you foresee any drawbacks with this approach? I would presume switching provider types per storage would be problematic either way?

Reply:
Yes, I think we three discussed that together. Switching the storage type in edit mode would only make sense if all or most of the storage attributes would stay the same. This is not the case here. Most of the fields would need to get reset once the storage type changes. Then I believe it is better to tell the user to start a storage from scratch.